solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java

```diff
@@ -271,17 +340,24 @@ public void transform(SolrDocument doc, int docid)
 
     private void implTransform(SolrDocument doc, int docid, Float score)
         throws IOException {
-      Object fv = featureLogger.getFeatureVector(docid, scoringQuery, searcher);
-      if (fv == null) { // FV for this document was not in the cache
-        fv = featureLogger.makeFeatureVector(
-            LTRRescorer.extractFeaturesInfo(
-                modelWeight,
-                docid,
-                (docsWereNotReranked ? score : null),
-                leafContexts));
+      LTRScoringQuery rerankingQuery = rerankingQueries[0];
+      LTRScoringQuery.ModelWeight rerankingModelWeight = modelWeights[0];
+      if (rerankingQueries.length > 1 && ((LTRInterleavingScoringQuery)rerankingQueries[1]).getPickedInterleavingDocIds().contains(docid)) {
```

Comment:
LTRFeatureLoggerTransformerFactory.7 - here now the reranking queries that were copied from the context (and modified) are being used. It's very subtle but questions that may arise for code readers are (a) whether or not the copy happened before or after the picked interleaving doc ids were set, and/or (b) if the copy is shallow so that the query from the context and the query in the transformer share the same underlying picked ids object, and/or (c) if any shared underlying picked ids object remains shared when 'set' is called on it. https://github.com/cpoerschke/lucene-solr/commit/3a61287a0e4fb5a77f080a92e7129582b234cbd7 proposes to introduce (alongside the `rerankingQueries` array) a `rerankingQueriesFromContext` array which is used only for the "picked doc id" check which would avoid the "was it copied before or after the picked ids were set" questions. An extra implementation subtlety is as follows: get-picked-interleaving-doc-ids applies only to interleaving queries and because of that the "copy-and-modify" approach was required for `rerankingQueries` before but if the already existing queries from the context can be used for the picked-or-not check then the `rerankingQueries` can follow a "construct-a-new-one" approach instead with the constructed queries all being of the same `LTRScoringQuery` base class type.

Reply:
Perfectly splendid! I agree!